inc/datahandlers/user.php

```diff
@@ -1783,4 +1783,61 @@ function clear_profile($delete_uids=false, $gid=0)
 			remove_avatars($uid);
 		}
 	}
+
+	public function verify_signature()
+	{
+		global $mybb, $parser;
+
+		$parser_options = array(
+			'allow_html' => $mybb->settings['sightml'],
+			'filter_badwords' => 1,
+			'allow_mycode' => $mybb->settings['sigmycode'],
+			'allow_smilies' => $mybb->settings['sigsmilies'],
+			'allow_imgcode' => $mybb->settings['sigimgcode'],
+			"filter_badwords" => 1
+		);
+
+		if($this->data['showimages'] != 1 && $this->data['uid'] != 0)
```

Comment:
I see you edited it while I was editing my comment. But now that I look at it, the conditional most likely was/is entirely wrong from the beginning, is probably another result of careless copy+paste and causes another much more serious bug... First of all, guests can't access UCP or have signature in the first place, so the 2nd part was entirely useless (it's useless here too: https://github.com/dequeues/mybb/blob/fix2298/usercp.php#L1953). Secondly, it uses current user's setting to overwrite the img code parser, which makes it possible for the user to have unlimited amount of images and ignore the group limit if they disable the `showimages` option... The conditional should be entirely deleted and `showimages` not passed.